src/AppBundle/Controller/Utils/AdminDashboardTrait.php

```diff
@@ -320,9 +322,16 @@ public function downloadTaskImage($taskId, $imageId, StorageInterface $storage)
 
         $response->setContentDisposition(
             ResponseHeaderBag::DISPOSITION_ATTACHMENT,
-            basename($imagePath)
+            $this->getImageDownloadFileName($image, $slugify)
         );
 
         return $response;
     }
+
+    protected function getImageDownloadFileName(TaskImage $taskImage, SlugifyInterface $slugify)
+    {
+        $taskImageNamer = new TaskImageNamer();
```

Comment:
I prefer constructor injection, but it's because I'm an object-oriented psycho 🤪
I mean, `getImageDownloadFileName($taskImage)` interface method is like "I give you an object, give me it's filename". The fact that it internally uses some object to slugify is implementation details.

```
$taskImageNamer = new TaskImageNamer($slugify);
return $taskImageNamer->getImageDownloadFileName($taskImage);
```

But there are no interfaces, it's overkill, and totally fine like this too 👍